tfutils/data.py

```diff
@@ -194,68 +192,93 @@ def isin(X,Y):
-                                               min_after_dequeue=n_threads * batch_size,
+            self.queue = tf.RandomShuffleQueue(capacity=self.capacity,
+                                               min_after_dequeue=self.capacity // 2,
                                                dtypes=dtypes,
                                                shapes=shapes,
-                                               names=node.keys(),
+                                               names=nodes.keys(),
                                                seed=seed)
         elif queue_type == 'fifo':
-            self.queue = tf.FIFOQueue(capacity=n_threads * batch_size * 2,
+            self.queue = tf.FIFOQueue(capacity=self.capacity,
                                       dtypes=dtypes,
                                       shapes=shapes,
-                                      names=node.keys())
+                                      names=nodes.keys())
         elif queue_type == 'padding_fifo':
-            self.queue = tf.PaddingFIFOQueue(capacity=n_threads * batch_size * 2,
+            self.queue = tf.PaddingFIFOQueue(capacity=self.capacity,
                                              dtypes=dtypes,
                                              shapes=shapes,
-                                             names=node.keys())
+                                             names=nodes.keys())
         elif queue_type == 'priority':
-            self.queue = tf.PriorityQueue(capacity=n_threads * batch_size * 2,
+            self.queue = tf.PriorityQueue(capacity=self.capacity,
                                           types=dtypes,
                                           shapes=shapes,
-                                          names=node.keys())
+                                          names=nodes.keys())
         else:
             Exception('Queue type %s not recognized' % queue_type)
 
-        if self.data_iter.batch_size > 1:
-            self.enqueue_op = self.queue.enqueue_many(node)
+        if data_batch_size > 1:
+            self.enqueue_op = self.queue.enqueue_many(nodes)
         else:
-            self.enqueue_op = self.queue.enqueue(node)
+            self.enqueue_op = self.queue.enqueue(nodes)
         self.batch = self.queue.dequeue_many(batch_size)
 
     def thread_main(self, sess):
         """
         Function run on alternate thread. Basically, keep adding data to the queue.
         """
-        for batch in self.data_iter:
-            if self._continue:
-                sess.run(self.enqueue_op, feed_dict=batch)
-            else:
-                break
+        try:
+            feed_dict = {node: self._first_batch[name] for name, node in self.nodes.items()}
+            sess.run(self.enqueue_op, feed_dict=feed_dict)
+        except tf.errors.CancelledError:
+            pass
+        else:
+            for batch in self.data_iter:
+                try:
```

Comment:
If you look at __init__, you'll see that I get the first batch from the data iterator because I need that first batch to get the shape of data inputs to construct placeholders. So here I first feed in this first batch and if that raises no errors, we continue onto the loop. Technically, the loop doesn't have to be under else, not sure why I put it there.